src/language.rs

```diff
@@ -95,81 +32,79 @@ pub static ALL_SUPPORTED_LANGUAGES: BySupportedLanguage<SupportedLanguage> = {
         Json,
         Css,
         Lua,
-    ]
-};
+    ],
+}
 
-static SUPPORTED_LANGUAGE_LANGUAGES: Lazy<BySupportedLanguage<Language>> = Lazy::new(|| {
-    [
-        tree_sitter_rust::language(),
-        tree_sitter_typescript::language_tsx(),
-        tree_sitter_javascript::language(),
-        tree_sitter_swift::language(),
-        tree_sitter_objc::language(),
-        tree_sitter_toml::language(),
-        tree_sitter_python::language(),
-        tree_sitter_ruby::language(),
-        tree_sitter_c::language(),
-        tree_sitter_cpp::language(),
-        tree_sitter_go::language(),
-        tree_sitter_java::language(),
-        tree_sitter_c_sharp::language(),
-        tree_sitter_kotlin::language(),
-        tree_sitter_elisp::language(),
-        tree_sitter_elm::language(),
-        tree_sitter_dockerfile::language(),
-        tree_sitter_html::language(),
-        tree_sitter_query::language(),
-        tree_sitter_json::language(),
-        tree_sitter_css::language(),
-        tree_sitter_lua::language(),
-    ]
-});
+impl SupportedLanguage {
+    pub fn language(&self) -> Language {
+        SUPPORTED_LANGUAGE_LANGUAGES[*self]
+    }
 
-#[inline(always)]
-pub fn get_supported_language_language(supported_language: SupportedLanguage) -> Language {
-    SUPPORTED_LANGUAGE_LANGUAGES[supported_language]
+    pub fn name_for_ignore_select(&self) -> &'static str {
+        SUPPORTED_LANGUAGE_NAMES_FOR_IGNORE_SELECT[*self]
+    }
 }
 
-pub static SUPPORTED_LANGUAGE_NAMES_FOR_IGNORE_SELECT: BySupportedLanguage<&'static str> = [
-    "rust",
-    "ts",
-    "js",
-    "swift",
-    "objc",
-    "toml",
-    "py",
-    "ruby",
-    "c",
-    "cpp",
-    "go",
-    "java",
-    "csharp",
-    "kotlin",
-    "elisp",
-    "elm",
-    "docker",
-    "html",
-    "treesitterquery",
-    "json",
-    "css",
-    "lua",
-];
+static SUPPORTED_LANGUAGE_LANGUAGES: Lazy<BySupportedLanguage<Language>> = Lazy::new(|| {
+    by_supported_language!(
```

Comment:
I think I sort of like this approach of exposing a secondary macro for allowing you to create new "fixed-size collections" (as opposed to eg specifying all of the `tree_sitter::Language`'s and "names for ignore select" in the body of the main `fixed_map!()` invocation and then either generating the same thing as this under the hood or possible generating "fat `SupportedLanguage` structs" (like how it used to have `language`/`name_for_ignore_select` as struct fields rather than these separate top-level data structures))?

I guess the downsides could be (a) in terms of code readability if one thought it were clearer to read "one big declaration" of all of the info/"fields" for all of the supported languages and (b) in terms of runtime if for some reason it were more performant ("cache locality"...?) (and if this were then actually a practical performance issue, I'm really just theorizing) to actually have those fields co-located as part of their relevant "`SupportedLanguage` struct" (vs being co-located with other instances of "that same field" like this)?